dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.core.models.Region;
+import uk.ac.ebi.eva.dbsnpimporter.models.SubSnpCoreFields;
+import uk.ac.ebi.eva.dbsnpimporter.sequence.ReadSequenceException;
+import uk.ac.ebi.eva.dbsnpimporter.sequence.SequenceReader;
+
+public class AssemblyCheckFilterProcessor implements ItemProcessor<SubSnpCoreFields, SubSnpCoreFields> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AssemblyCheckFilterProcessor.class);
+
+    private SequenceReader sequenceReader;
+
+    public AssemblyCheckFilterProcessor(SequenceReader assemblyReader) {
+        this.sequenceReader = assemblyReader;
+    }
+
+    @Override
+    public SubSnpCoreFields process(SubSnpCoreFields subSnpCoreFields) throws Exception {
+        String referenceAllele = subSnpCoreFields.getReferenceInForwardStrand();
+        // check if reference is not empty
+        if (isEmpty(referenceAllele)) {
+            return subSnpCoreFields;
+        } else {
+            Region region = subSnpCoreFields.getVariantCoordinates();
+            try {
+                String sequenceInAssembly = sequenceReader.getSequence(region.getChromosome(), region.getStart(),
+                                                                       region.getEnd());
+                if (referenceAllele.equals(sequenceInAssembly)) {
+                    return subSnpCoreFields;
+                } else {
+                    logger.warn("Variant filtered out because it the reference allele is not correct: {}.\n" +
+                                        "{} expected in {}, {} found",
+                                subSnpCoreFields, sequenceInAssembly, region, referenceAllele);
+                    return null;
+                }
+            } catch (ReadSequenceException e) {
```

Comment:
This can be thrown also if end is greater than start, or if the coordinates are outside the chromosome limits. If we throw the exception and a variant is stored in dbSNP with incorrect coordinates, we are not going to be able to continue the execution for that batch unless we change the code. I'm going to try to separate this in different exceptions, so we stop the execution just if the fasta read cannot be read.

Reply:
Yes, in a comment below I suggest using `IllegalArgumentException` in the case of illegal coordinates being provided.